src/main/java/org/junit/runners/ParentRunner.java

```diff
@@ -206,6 +209,29 @@ private boolean areAllChildrenIgnored() {
     }
 
     /**
+     * Returns a {@link Statement}: notify all non-overridden {@code @Ignore} methods on this class
+     * and superclasses if the given {@code statement} throws an exception.
+     */
+    protected Statement withNotificationOfIgnoredChildren(final Statement statement,
+            final RunNotifier notifier) {
+        return new Statement() {
+            @Override
+            public void evaluate() throws Throwable {
+                try {
+                    statement.evaluate();
+                } catch (Throwable e) {
+                    runChildren(notifier, new ChildSelector<T>() {
+                        public boolean include(T child) {
+                            return isIgnored(child);
+                        }
+                    });
+                    throw e;
```

Comment:
In the catch part, calling runChildren() could raise a second exception, which should be catched, too. Otherwise the information of the previous catched exception is lost.